components/cam/src/dynamics/se/stepon.F90

```diff
@@ -414,6 +455,14 @@ subroutine stepon_run2(phys_state, phys_tend, dyn_in, dyn_out )
       end do
    endif
    
+   do ie = 1,nelemd
```

Comment:
Related to the point above, these outfld calls appear new and active even in the baseline code path. Intended?

Reply:
see my response above. I don't think these have any performance implications as long as the fields aren't requested to be in the history files.

Reply:
I don't know this section of code well enough to make a recommendation. I brought up the difference only because I observed it.

Reply:
Just to complete the conversation to Walter's point.  In order for a field to actually be written to file it needs to be added via namelist or more generally added as "default" using the add_default command.  I can't speak here to whether or not there is any performance hit to calling outfld even if the field is not actually going to be written, I think maybe @worleyph could speak to that.  But I think at this point this approach shouldn't add any more outfield data unless it is specified by namelist.

Reply:
I think if their was a performance concern here then the amount of addfld calls would be limited. Right now there are A LOT on the physics side just in case a user wants those variables.

Reply:
You would think that.  But if it were the case that one little outfld call wasn't much of an issue you could certainly see a death by a thousand cuts situation where each new outfld call isn't a terrible thing but they add up.  That wouldn't be a reason to focus just on these 6, it would speak to what should be a broader effort to revamp a code riddled with outfld calls that are hardly used.  So, I think the broader question is what are the performance impacts of calling outfld?  Is there any reason to go through the code and really clean these up?

Reply:
I think it's probably a good idea to go through these. I think they could be a bit more "centralized" as well. Additionally, it would be good to just make a new module on the dynamics side to encapsulate them all. But that's a low priority for another PR.